toolsrc/src/vcpkg/configuration.cpp

```diff
@@ -28,17 +30,24 @@ namespace vcpkg
             }
         }
 
-        static Json::ArrayDeserializer<RegistryDeserializer> array_of_registries{"an array of registries"};
-
+        auto reg_des = get_registry_array_deserializer(configuration_directory);
         std::vector<Registry> regs;
-        r.optional_object_field(obj, REGISTRIES, regs, array_of_registries);
+        r.optional_object_field(obj, REGISTRIES, regs, *reg_des);
 
         if (!regs.empty() && !registries_enabled)
         {
             registries_feature_flags_warning = true;
             regs.clear();
         }
 
+        if (!r.errors().empty())
+        {
+            System::print2(System::Color::error, "Errors occurred while parsing configuration.\n");
+            for (auto&& msg : r.errors())
+                System::print2("    ", msg, '\n');
+            Checks::exit_fail(VCPKG_LINE_INFO);
```

Comment:
We should not be printing nor terminating during any deserializers; this makes it painful to test and prevents composition.
```suggestion
            return Configuration{};
```